extensions/sql-migration/src/main.ts

```diff
@@ -76,16 +81,47 @@ class SQLMigration {
 	async launchMigrationWizard(): Promise<void> {
 		let activeConnection = await azdata.connection.getCurrentConnection();
 		let connectionId: string = '';
+		let serverName: string = '';
 		if (!activeConnection) {
 			const connection = await azdata.connection.openConnectionDialog();
 			if (connection) {
 				connectionId = connection.connectionId;
+				serverName = connection.options.server;
 			}
 		} else {
 			connectionId = activeConnection.connectionId;
+			serverName = activeConnection.serverName;
+		}
+		if (serverName) {
+			const api = (await vscode.extensions.getExtension(mssql.extension.name)?.activate()) as mssql.IExtension;
+			if (api) {
+				this.stateModel = new MigrationStateModel(this.context, connectionId, api.sqlMigration);
+				this.context.subscriptions.push(this.stateModel);
+				let savedInfo = this.checkSavedInfo(serverName);
+				if (savedInfo) {
+					this.stateModel.savedInfo = savedInfo;
+					this.stateModel.serverName = serverName;
+					let savedAssessmentDialog = new SavedAssessmentDialog(this.context, this.stateModel);
+					await savedAssessmentDialog.openDialog();
+				} else {
+					const wizardController = new WizardController(this.context, this.stateModel);
+					await wizardController.openWizard(connectionId);
+				}
+			}
+
+		}
+
+
+
+	}
+
+	private checkSavedInfo(serverName: string): SavedInfo | undefined {
+		let savedInfo: SavedInfo | undefined = this.context.globalState.get(`${this.stateModel.mementoString}.${serverName}`);
+		if (savedInfo) {
+			return savedInfo;
+		} else {
+			return;
```

Comment:
`return undefined` to be explicit from returning void

(but you can also just return the call to get directly since that already returns an object or undefined)